app.js

```diff
@@ -19,15 +19,23 @@ var bot = Slackbot(require('@slack/client'), env);
 var tweeter = TwitterPoster(env);
 var log = helpers.logger('App', env.isDev);
 
+function onImageDataDone (err, imageData, trimmedText) {
+  if (err) {
+    throw err;
+  }
+  console.log('callback called!');
+  tweeter.postTweetAndImage(imageData, trimmedText);
+}
+
 function postText (message) {
+  console.log(message);
   var text = message.item.message && message.item.message.text;
   if (text && text.length <= 140) {
     tweeter.postTweet(text);
   } else if (text && text.length > 140) {
     log('generating tweet for text > 140 characters');
     var trimmedText = text.substring(0, 110) + '…';
-    var imageData = createFakeScreenshot(text);
-    tweeter.postTweetAndImage(imageData, trimmedText);
+    createFakeScreenshot(message.item.message.permalink, trimmedText, onImageDataDone);
```

Comment:
also, handle errors with `helpers.handleError` instead of throwing them. now that we're using promises, `throw` doesn't work like we want it to anymore. i'm gonna have to change my implementation of `helpers.handleError`, so we want all error handling logic in the same place.